src/services/tx/txSender.ts

```diff
@@ -57,7 +59,23 @@ export const createTx = async (txParams: SafeTransactionDataPartial, nonce?: num
   // Set the recommended nonce and safeTxGas if not provided
   if (nonce === undefined) {
     const chainId = await safeSDK.getChainId()
-    const estimation = await estimateSafeTxGas(String(chainId), safeSDK.getAddress(), txParams)
+    const safeNonce = await safeSDK.getNonce()
+    const safeAddress = safeSDK.getAddress()
+    let estimation: SafeTransactionEstimation
+    try {
+      estimation = await estimateSafeTxGas(String(chainId), safeAddress, txParams)
+    } catch (e) {
+      try {
+        /* If the initial transaction data causes the estimation to fail
+         we retry the request with a cancellation transaction to get the
+         recommendedNonce even if the original transaction will likely fail */
+        const cancellationTxParams = { ...txParams, data: EMPTY_DATA, to: safeAddress, value: '0' }
+        estimation = await estimateSafeTxGas(String(chainId), safeAddress, cancellationTxParams)
+      } catch (e) {
+        logError(Errors._616, (e as Error).message)
+        estimation = { currentNonce: safeNonce, recommendedNonce: safeNonce, safeTxGas: '0' }
+      }
+    }
     txParams = { ...txParams, nonce: estimation.recommendedNonce, safeTxGas: Number(estimation.safeTxGas) }
```

Comment:
I would leave the estimation as undefined in the final catch, and not modify the txParams in that case.
The txParams should already contain the current safe nonce, so you can remove `const safeNonce = await safeSDK.getNonce()` too.

Reply:
That works well! Just tried to remove the first catch to enter directly in this case and it is handled elegantly 👌 

👇 
```ts
try {
  estimation = await estimateSafeTxGas(String(chainId), safeAddress, txParams)
} catch (e) {
  logError(Errors._616, (e as Error).message)
  return safeSDK.createTransaction({ safeTransactionData: txParams })
}
```

Reply:
Just pushed then change